src/views/SpaceSettings.vue

```diff
@@ -287,6 +297,48 @@ watch(
   }
 );
 
+function handleEnsButtonClick() {
+  if (isOwner.value || isAdmin.value) {
+    window.open(`https://app.ens.domains/name/${props.spaceId}`, '_blank');
+    return;
+  }
+
+  handleSetRecord();
+}
+
+async function handleSetRecord() {
+  settingENSRecord.value = true;
+  try {
+    const ensPublicResolverAddress = networks[networkKey.value].ensResolver;
+    if (!ensPublicResolverAddress) {
+      throw new Error('No ENS resolver address for this network');
+    }
+    const ensname = props.spaceId;
+    const node = namehash.hash(ensname);
+    const tx = await sendTransaction(
+      auth.web3,
+      ensPublicResolverAddress,
+      abi,
+      'setText',
+      [node, 'snapshot', textRecord.value]
+    );
+    const receipt = await tx.wait(2);
```

Comment:
1 confirmation should be fine. The finality difference between 1 and 2 confirmations isn't that much. 12 would be considered 99.99% final.

https://blog.ethereum.org/2015/09/14/on-slow-and-fast-block-times/

Then also we are just talking about a UI's responsiveness. What I mean is... once the user wants to save the settings there will be like 100 confirmations. And after all, worst case is that there's no text record and the user needs to set it again. Can't think of any issues here.

```suggestion
    const receipt = await tx.wait();
```